### ISSUE TITLE
Handle `IS [NOT] DISTINCT FROM` in some parsers

### ISSUE THREAD

**Issue by leamingrad** [sqlfluff/sqlfluff#1487]
## Expected Behaviour

The following file should parse with the `postgres` dialect:

```sql
-- Column distinctness in SELECT expression
SELECT a_column IS DISTINCT FROM b_column FROM t_table;
SELECT b_column IS NOT DISTINCT FROM c_column FROM t_table;

-- Column distinctness from constant in SELECT expression
SELECT c_column IS DISTINCT FROM 'test' FROM t_table;
SELECT c_column IS NOT DISTINCT FROM 'test' FROM t_table;

-- Constant distinctness
SELECT 4 IS DISTINCT FROM 5;
SELECT 4 IS NOT DISTINCT FROM 5;

-- Column distinctness in WHERE expression
SELECT * FROM t_table WHERE a_column IS DISTINCT FROM b_column;
SELECT * FROM t_table WHERE a_column IS NOT DISTINCT FROM b_column;

-- Column distinctness from constant in WHERE expression
SELECT * FROM t_table WHERE a_column IS DISTINCT FROM 211;
SELECT * FROM t_table WHERE a_column IS NOT DISTINCT FROM 211;
```

## Observed Behaviour

`sqlfluff parse --dialect postgres` gives the following parsing violations (full output attached):

```
==== parsing violations ====
L:   2 | P:  16 |  PRS | Line 2, Position 16: Found unparsable section: ' IS DISTINCT'
L:   2 | P:  43 |  PRS | Line 2, Position 43: Found unparsable section: 'FROM t_table'
L:   3 | P:  16 |  PRS | Line 3, Position 16: Found unparsable section: ' IS NOT DISTINCT'
L:   3 | P:  47 |  PRS | Line 3, Position 47: Found unparsable section: 'FROM t_table'
L:   6 | P:  16 |  PRS | Line 6, Position 16: Found unparsable section: ' IS DISTINCT'
L:   6 | P:  29 |  PRS | Line 6, Position 29: Found unparsable section: "FROM 'test' FROM
                       | t_table"
L:   7 | P:  16 |  PRS | Line 7, Position 16: Found unparsable section: ' IS NOT DISTINCT'
L:   7 | P:  33 |  PRS | Line 7, Position 33: Found unparsable section: "FROM 'test' FROM
                       | t_table"
L:  10 | P:   9 |  PRS | Line 10, Position 9: Found unparsable section: ' IS DISTINCT'
L:  10 | P:  22 |  PRS | Line 10, Position 22: Found unparsable section: 'FROM 5'
L:  11 | P:   9 |  PRS | Line 11, Position 9: Found unparsable section: ' IS NOT DISTINCT'
L:  11 | P:  26 |  PRS | Line 11, Position 26: Found unparsable section: 'FROM 5'
L:  14 | P:  37 |  PRS | Line 14, Position 37: Found unparsable section: ' IS DISTINCT FROM
                       | b_column'
L:  15 | P:  37 |  PRS | Line 15, Position 37: Found unparsable section: ' IS NOT DISTINCT
                       | FROM b_column'
L:  18 | P:  37 |  PRS | Line 18, Position 37: Found unparsable section: ' IS DISTINCT FROM
                       | 211'
L:  19 | P:  37 |  PRS | Line 19, Position 37: Found unparsable section: ' IS NOT DISTINCT
                       | FROM 211'
```

[error_output.txt](https://github.com/sqlfluff/sqlfluff/files/7268658/error_output.txt)

## Dialect

This syntax appears to be non-standard, but as far as I can tell it is supported by:
* Postgres - https://www.postgresql.org/docs/11/functions-comparison.html
* Snowflake - https://docs.snowflake.com/en/sql-reference/functions/is-distinct-from.html
* Exasol - https://www.jooq.org/doc/3.3/manual-single-page/ (I haven't heard of Exasol before so I am not sure if this is the correct manual)

## Version
Reproduced with checkout of `main` on Python 3.8.3

**Comment by WittierDinosaur** [comment]
Looking at the docs - this is a comparison operator. (INFO)

**Comment by WittierDinosaur** [comment]
Addressed in Postgres in #1818

**Comment by tunetheweb** [comment]
I’m going to close this. It was fixed for Postgres (the original ask). We can open other issues if people want it for other dialects.

**Comment by alexscott-ff** [comment]
Hey @tunetheweb would love this for Snowflake as well if possible, just running into this today.

**Comment by dmohns** [comment]
Hey, I would appreciate this for BigQuery as well. It's supported in BigQuery https://cloud.google.com/bigquery/docs/reference/standard-sql/operators#is_distinct Should be create new issue for this?

**Comment by tunetheweb** [comment]
Yes if you wouldn't mind please.

Adding it outside ot the `SELECT` clause looks easy. Allowing it inside the `SELECT` clause is a bit more difficult as we use `FROM` as a signal to end the `SELECT` clause and start the `FROM` clause (not unreasonably). We would allow it in side parenthesis though. Can't see an easy way of doing this so the BigQuery examples don't work. But not sure how common that use case is.

```sql
-- This can be made to work:
SELECT (1 IS NOT DISTINCT FROM 2)

-- This cannot be made to work at the moment AFAIK:
SELECT 1 IS NOT DISTINCT FROM 2
```

Not sure if that would unblock you? IF you give a sample SQL you want to parse in the new issue you open we could see.
